### PR TITLE
Allow parallel vnode initialization

### PR DESCRIPTION
**Pull request by engelsanchez**
Solution to basho/platform_tasks#18

### CODE REVIEW

File: src/riak_core_util.erl [hunk 1]
```diff
@@ -454,5 +530,72 @@ build_tree_test() ->
...
+        {counter_value, N} -> N
+    after
+        3000 ->
+            ?assert(false)
+    end.
+
+decr_counter(CounterPid) ->
+    CounterPid ! down.
+
+bounded_pmap_test_() ->
```
Comment (jonmeredith) [inline src/riak_core_util.erl:558]: Should #ifdef(TEST) around this

Comment (engelsanchez) [inline src/riak_core_util.erl:558]: It was added to an existing ifdef TEST region, so it doesn't show up in the diff, but it is.

Comment (jonmeredith) [inline src/riak_core_util.erl:558]: I think I'll make some phone calls or call a meeting or something.

Comment (engelsanchez) [inline src/riak_core_util.erl:558]: Don't give into the dark side Jon!!!!!! :)
http://i.livescience.com/images/i/000/020/685/iFF/luke-skywalker-111005.jpg?1317834193

**Comment by evanmcc** [comment]
do we have any benchmarks or assessments of performance impact?

**Comment by engelsanchez** [comment]
Only on a developer laptop with an SSD, 16 partitions and 12GB of bitcask data. The riak_kv service took 4min+ to initialize without this, but only 1min + a few seconds with this change. All coming from being able to better saturate the I/O with concurrent work.  This will be tested on real clusters ASAP. If you want to lend a hand, let me know. I know you have something set up and still warm that perhaps we could use?  With the latest commit, now concurrency can be controlled with the previously existing vnode_rolling_start parameter. The default should be good (16), and you would get the old, sequential behavior if set to 1 or less.  Tweaking on rotational drives should be interesting as I would expect it to turn around and get worse as you tune that setting up. I'm dying to see the real effect.

**Comment by evanmcc** [comment]
code looks good

testing with 16 vnodes containing 320GB of data on a rotational disk raid system indicated a 10X speedup.

:+1:

**Pull request merged**